Content/Infrastructure/Sulu/SmartContent/Repository/ContentDataProviderRepository.php

```
@@ -113,7 +113,7 @@ function (ContentRichEntityInterface $contentRichEntity) use ($locale, $showUnpu
                     $dimension = $resolvedDimensionContent->getDimension();
 
                     if ($stage !== $dimension->getStage() || $locale !== $dimension->getLocale()) {
-                        return null;
+                        return null; // @codeCoverageIgnore
```

I'm not sure if this if statement is needed we query for `stage` and the correct `locale` in the queryBuilder. When here the false stage and locale would be returned it would be a bug in the `contentResolver` and should throw an exception but I don't see any need for exception or check here. 

@nnatter would you remove the if else statement? /cc @luca-rath any knowledge why it was added?

---

I'm not sure if it's still the case, but at some point it was possible to query for `['locale' => 'en', 'stage' => 'live']` and the result was a `DimensionContent` with these `['locale' => null, 'stage' => 'live']` dimension attributes ... This happened, if the entity was not published in the requested locale but was published in any other locale

---

Thx for feedback this should not happen because of the following where statement: https://github.com/sulu/SuluContentBundle/blob/0e9ebd59156c19b5469f3706fb89b8af0216e7b9/Content/Infrastructure/Sulu/SmartContent/Repository/ContentDataProviderRepository.php#L453

If this would really be the case we would need to have a look again at that query, or the contentResolver / merger.

---

I think it is still the case. See https://github.com/sulu/SuluContentBundle/issues/145

---

@nnatter the query itself should not return any `ids` which has that so #145 should not happen here.

---

Okay thats possible - have not looked into the implementation of the `ContentDataProviderRepository` 🙂

---

I guess it would make sense to introduce a new argument like $allowPartial or something like that for the `ContentResolver::resolve()`, so the ContentResolver can make that above check for you and throw an exception if it fails.
Because sometimes it would totally make sense, if I get only a partial merged DimensionContent from the ContentResolver. An example would be, if I switch to a new locale in the admin, the unlocalized fields should already be prefilled, even if that entity doesn't exist in that locale.

---

@luca-rath we will definitely add an options array as things like `load_ghost` and `load_shadow` need to be handled in future over the resolver. So I would not introduce here a specific boolean flag. But thats part of #145